src/state/collectibles/index.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+import { CollectiblesState } from 'state/types'
+import { nftSources } from 'config/constants/nfts'
+import { NftType } from 'config/constants/types'
+import { getAddress } from 'utils/addressHelpers'
+import { getErc721Contract } from 'utils/contractHelpers'
+import { getNftByTokenId } from 'utils/collectibles'
+
+const initialState: CollectiblesState = {
+  isInitialized: false,
+  isLoading: true,
+  data: {},
+}
+
+type NftSourceItem = [number, string]
+
+// Thunks
+export const fetchWalletNfts = createAsyncThunk<NftSourceItem[], string>(
+  'collectibles/fetchWalletNfts',
+  async (account) => {
+    // For each nft source get nft data
+    const nftSourcePromises = Object.keys(nftSources).map(async (nftSourceType) => {
+      const { address: addressObj } = nftSources[nftSourceType as NftType]
+      const address = getAddress(addressObj)
+      const contract = getErc721Contract(address)
+
+      const getTokenIdAndData = async (index: number) => {
+        try {
+          const tokenId = await contract.methods.tokenOfOwnerByIndex(account, index).call()
```

Comment:
this is still poorly optimised because we will do one request for each object. Instead we should batch the requests